File: src/main/rs/emulate/legacy/graphics/sprite/SpriteEncoder.java
package rs.emulate.legacy.graphics.sprite;

import java.util.Arrays;
import java.util.List;

import rs.emulate.legacy.archive.Archive;
import rs.emulate.legacy.graphics.GraphicsEncoder;

/**
 * Encodes a {@link Sprite} into an archive entry. TODO
 * 
 * @author Major
 */
public final class SpriteEncoder extends GraphicsEncoder {

	/**
	 * The List of Sprites to encode.
	 */
	@SuppressWarnings("unused")
	private final List<Sprite> sprites;

	/**
	 * Creates the SpriteEncoder.
	 * 
	 * @param name The name of the {@link Sprite}.
	 * @param sprites The Sprites to encode.
	 */
	public SpriteEncoder(String name, Sprite... sprites) {
		super(name);
		this.sprites = Arrays.asList(sprites);
	}

	@Override
	public Archive encodeInto(Archive archive) {
		return null;
	}

}

File: legacy/src/main/kotlin/rs/emulate/legacy/graphics/sprite/SpriteEncoder.kt
package rs.emulate.legacy.graphics.sprite

import rs.emulate.legacy.archive.Archive
import rs.emulate.legacy.graphics.GraphicsEncoder

import java.util.Arrays

/**
 * Encodes a [Sprite] into an archive entry. TODO
 *
 * @param name The name of the [Sprite].
 * @param sprites The Sprites to encode.
 */
class SpriteEncoder(name: String, vararg sprites: Sprite) : GraphicsEncoder(name) {

    private val sprites: List<Sprite> = Arrays.asList(*sprites)

    override fun encodeInto(archive: Archive): Archive {
        TODO()
    }

}
